modules/playlists/src/main/java/org/opencastproject/playlists/Playlist.java

```diff
@@ -0,0 +1,240 @@
+   * hibernate version).
+   */
+  @PrePersist
+  public void generateId() {
+    if (this.id == null) {
+      this.id = UUID.randomUUID().toString();
+    }
+  }
+
+  @Id
+  @Column(name = "id")
+  private String id;
+
+  @Column(name = "organization", nullable = false, length = 128)
+  private String organization;
+
+  @OneToMany(
+      fetch = FetchType.LAZY,
+      cascade = { CascadeType.PERSIST, CascadeType.REMOVE },
+      mappedBy = "playlist",
+      orphanRemoval = true
+  )
+  @OrderColumn(name = "position_entries")
+  private List<PlaylistEntry> entries = new ArrayList<PlaylistEntry>();
+
+  @Column(name = "title")
+  private String title;
+
+  @Column(name = "description")
+  private String description;
+
+  @Column(name = "creator")
+  private String creator;
+
+  @Column(name = "updated", nullable = false)
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date updated;
+
+  @Column(name = "deletion_date")
+  @Temporal(TemporalType.TIMESTAMP)
+  protected Date deletionDate = null;
+
+  @OneToMany(
+      fetch = FetchType.LAZY,
+      cascade = { CascadeType.PERSIST, CascadeType.REMOVE },
+      mappedBy = "playlist",
+      orphanRemoval = true
+  )
+  @OrderColumn(name = "position_access_control_entries")
+  private List<PlaylistAccessControlEntry> accessControlEntries = new ArrayList<>();
+
+  /**
+   * Default constructor
+   */
+  public Playlist() {
+
+  }
+
+  public Playlist(String id, String organization, List<PlaylistEntry> entries, String title, String description,
+      String creator, Date updated, List<PlaylistAccessControlEntry> accessControlEntries) {
+    this.id = id;
+    this.organization = organization;
+    this.entries = entries;
+    this.title = title;
+    this.description = description;
+    this.creator = creator;
+    this.updated = updated;
+    this.accessControlEntries = accessControlEntries;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public String getOrganization() {
+    return organization;
+  }
+
+  public void setOrganization(String organization) {
+    this.organization = organization;
+  }
+
+  public List<PlaylistEntry> getEntries() {
+    return entries;
+  }
+
+  public void setEntries(List<PlaylistEntry> entries) {
+    for (var entry : entries) {
+      if (!entries.stream().anyMatch(e -> entry.getId() == e.getId())) {
+        entry.setId(0L);
+      }
+      entry.setPlaylist(this);
+    }
+    this.entries = entries;
+  }
+
+  public boolean addEntry(PlaylistEntry entry) {
+    notNull(entry, "entry");
+    if (this.entries.contains(entry)) {
+      entries.remove(entry);
+    }
+    entry.setPlaylist(this);
+    return entries.add(entry);
+  }
+
+  public boolean removeEntry(PlaylistEntry entry) {
+    notNull(entry, "entry");
+    entry.setPlaylist(null);
+    return entries.remove(entry);
+  }
```

Comment:
This will only work if the equals method is overwritten. Maybe @LukasKalbertodt has an opinion about equality as he was writing the specification, or else remove them and decide later because they are not used anywhere. Do you think the same PK is enough?

The example below assumes an entry is equal if the PK is the same; if the PK is 0, they are not considered the same because they will get a different PK after persisting in the database.
  
  ```java
  @Test
  public void addEntryShouldAddNewEntryWhenNotAlreadyPresent() throws Exception {
    Playlist playlist = new Playlist();
    PlaylistEntry playlistEntry = new PlaylistEntry(1, "1234", PlaylistEntryType.EVENT);

    boolean result = playlist.addEntry(playlistEntry);

    Assert.assertTrue(result);
    Assert.assertEquals(1, playlist.getEntries().size());
    Assert.assertEquals(playlistEntry, playlist.getEntries().get(0));
  }

  @Test
  public void addEntryShouldReplaceExistingEntry() throws Exception {
    Playlist playlist = new Playlist();
    PlaylistEntry playlistEntry = new PlaylistEntry(1, "1234", PlaylistEntryType.EVENT);
    playlist.addEntry(playlistEntry);

    PlaylistEntry newPlaylistEntry = new PlaylistEntry(1, "5678", PlaylistEntryType.EVENT);
    boolean result = playlist.addEntry(newPlaylistEntry);

    Assert.assertTrue(result);
    Assert.assertEquals(1, playlist.getEntries().size());
    Assert.assertEquals(newPlaylistEntry, playlist.getEntries().get(0));
  }

  @Test
  public void addEntryWithoutIdNotEqual() throws Exception {
    Playlist playlist = new Playlist();
    PlaylistEntry playlistEntry = new PlaylistEntry(0, "1234", PlaylistEntryType.EVENT);
    playlist.addEntry(playlistEntry);

    PlaylistEntry newPlaylistEntry = new PlaylistEntry(0, "5678", PlaylistEntryType.EVENT);
    boolean result = playlist.addEntry(newPlaylistEntry);

    Assert.assertTrue(result);
    Assert.assertEquals(2, playlist.getEntries().size());
  }

  @Test
  public void removeExistingEntry() throws Exception {
    Playlist playlist = new Playlist();
    PlaylistEntry playlistEntry = new PlaylistEntry(1, "1234", PlaylistEntryType.EVENT);
    playlist.addEntry(playlistEntry);

    PlaylistEntry removePlaylistEntry = new PlaylistEntry(1, "1234", PlaylistEntryType.EVENT);
    boolean result = playlist.removeEntry(removePlaylistEntry);

    Assert.assertTrue(result);
    Assert.assertEquals(0, playlist.getEntries().size());
  }

  @Test(expected = IllegalArgumentException.class)
  public void addEntryShouldThrowExceptionWhenEntryIsNull() throws Exception {
    Playlist playlist = new Playlist();
    playlist.addEntry(null);
  }
  ```
  
  PlaylistEntry equals & hashCode. 
  ```java
  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PlaylistEntry that = (PlaylistEntry) o;
    return getId() != 0 && getId() == that.getId();
  }

  @Override
  public int hashCode() {
    return Objects.hash(getId());
  }
  ```

Reply:
I'm not Lukas, but I'd argue the same PK to be enough and your tests make sense to me. I added your code to a new commit ([7db86e3](https://github.com/opencast/opencast/pull/5478/commits/7db86e3626854e8a11d58023c20d0be1474df725) while referencing you, hope that is ok (else I'll revert it).